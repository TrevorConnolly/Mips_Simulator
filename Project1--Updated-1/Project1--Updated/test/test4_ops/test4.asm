main:
  # Test SUBU
  li $t4, 30          # Load immediate value into $t4
  li $t5, 10          # Load immediate value into $t5
  subu $t6, $t4, $t5  # Subtract $t5 from $t4, store result in $t6
  # Expected: $t6 = 20 (30 - 10)

  # Test ADDU
  li $t7, 15          # Load immediate value into $t7
  li $t8, 25          # Load immediate value into $t8
  addu $t9, $t7, $t8  # Add $t7 and $t8, store result in $t9
  # Expected: $t9 = 40 (15 + 25)

  # Test ADD with negative numbers
  li $t2, -15         # Load immediate value into $t0
  li $t1, 10          # Load immediate value into $t1
  add $t3, $t2, $t1   # Add $t0 and $t1, store result in $t2
  # Expected: $t2 = -5 (-15 + 10)

  # Test SUB with negative numbers
  li $s2, -20        
  li $s1, -10         
  sub $s3, $s2, $s1   

  .word 0xfeedfeed