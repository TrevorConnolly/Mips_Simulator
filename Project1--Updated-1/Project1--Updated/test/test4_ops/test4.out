---------------------
Begin Register Values
---------------------
$at = 0x00000000

$v0 = 0x00000000
$v1 = 0x00000000

$a0 = 0x00000000
$a1 = 0x00000000
$a2 = 0x00000000
$a3 = 0x00000000

$t0 = 0x00000000
$t1 = 0x0000000A
$t2 = 0xFFFFFFF1
$t3 = 0xFFFFFFFB
$t4 = 0x0000001E
$t5 = 0x0000000A
$t6 = 0x00000014
$t7 = 0x0000000F
$t8 = 0x00000019
$t9 = 0x00000028

$s0 = 0x00000000
$s1 = 0xFFFFFFF6
$s2 = 0xFFFFFFEC
$s3 = 0xFFFFFFF6
$s4 = 0x00000000
$s5 = 0x00000000
$s6 = 0x00000000
$s7 = 0x00000000

$k0 = 0x00000000
$k1 = 0x00000000

$gp = 0x00000000
$sp = 0x00000000
$fp = 0x00000000
$ra = 0x00000000
---------------------
End Register Values
---------------------
